.github/workflows/test.yml

```diff
@@ -52,7 +52,7 @@ jobs:
     if: ${{ needs.changes.outputs.changes == 'true' }}
     strategy:
       matrix:
-        python-version: ["3.9", "3.10", "3.11"]
+        python-version: ["3.9", "3.11", "3.12"]
```

Comment:
I think it's fine to run code style in all versions?

Reply:
Actually, this seems very strange to me. Why would we run pre-commit on more than one Python version? Pre-commit provides environment isolation for each hook, so it shouldn't matter from what version you run it.

Reply:
It catches stuff like syntax that is not supported in older versions

Reply:
But I'm happy with just running on the oldest and most recent versions. Either that or all. A thing in between seems odd.